code/clue.js

```diff
@@ -1,59 +1,396 @@
+  weight: 1,
+  flexibility: 9,
+  color: '#d1a93c',
+  id: 'pistol',
+  image: 'assets/pistol.jpg'
+};
+
+// OBJECTS FOR THE ROOMS (BECAUSE THEY HAVE PICTURES)
+const diningRoom = {
+  name: 'Dining Room',
+  image: 'assets/dining.jpg'
+};
+
+const conservatory = {
+  name: 'Conservatory',
+  image: 'assets/conservatory.jpg'
+};
+
+const kitchen = {
+  name: 'Kitchen',
+  image: 'assets/kitchen.jpg'
+};
+
+const study = {
+  name: 'Study',
+  image: 'assets/study.jpg'
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const library = {
+  name: 'Library',
+  image: 'assets/library.jpg'
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const billiardRoom = {
+  name: 'Billiard Room',
+  image: 'assets/billiard.jpg'
+};
 
+const lounge = {
+  name: 'Lounge',
+  image: 'assets/lounge.jpg'
+};
+
+const ballRoom = {
+  name: 'Ball Room',
+  image: 'assets/ballroom.jpg'
+};
+
+const hall = {
+  name: 'Hall',
+  image: 'assets/hall.jpg'
+};
+
+const spa = {
+  name: 'Spa',
+  image: 'assets/spa.jpg'
+};
+
+const livingRoom = {
+  name: 'Living Room',
+  image: 'assets/livingroom.jpg'
+};
+
+const observatory = {
+  name: 'Observatory',
+  image: 'assets/observatory.jpg'
+};
+
+const theater = {
+  name: 'Theater',
+  image: 'assets/theater.jpg'
+};
+
+const guestHouse = {
+  name: 'Guest House',
+  image: 'assets/guesthouse.jpg'
+};
+
+const patio = {
+  name: 'Patio',
+  image: 'assets/patio.jpg'
+};
+
+// STEP 2. NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candleStick,
+  dumbBell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballRoom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+];
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+const randomSelector = array => { //I don't understand this
+  return array[Math.floor(Math.random() * array.length)] //What does array.length do here?
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY. With a killer, a weapon and a room. The values will be set later.
+const mystery = {
+  killer: "",
+  weapon: "",
+  room: "",
+};
+
+//Function to shuffle the favourite weapons of the suspects in a randomized way.
+const shuffleWeapons = () => {
+  suspects.forEach(suspect => {
+    suspect.favorite = randomSelector(weapons).name
```

Comment:
An alternative here would be to 
`suspect.favorite = randomSelector(weapons)`

This would give you access to the entire weapon object that this suspect prefers, not just the name.  Could come in handy later, but your solution is precise👍